rfcs/036-ml-feature-discovery.md

```diff
@@ -0,0 +1,195 @@
+- Feature Name: ml_feature_discovery
+- Start Date: 2021-05-05
+- RFC PR: [amundsen-io/rfcs#36](https://github.com/amundsen-io/rfcs/pull/36) (after opening the RFC PR, update this with a link to it and update the file name)
+- Amundsen Issue: [amundsen-io/amundsen#0000](https://github.com/amundsen-io/amundsen/issues/0000) (leave this empty for now)
+
+# ML Features Discovery in Amundsen
+
+## Summary
+
+Amundsen currently indexes a few first-class entities, however we currently do not index any ML features that are easily discoverable for users. The goal of this work is to index ML features at the same level as Tables, Dashboards, and People resources in Amundsen.
+
+## Motivation
+
+Integration of ML projects in Amundsen has been heavily requested by the community and has already been implemented by other similar projects. This RFC outlines what will be the first step towards indexing ML projects in Amundsen since feature engineering is one of the most time consuming processes for ML engineers. Providing them with the ability to easily discover features that they can reuse for their models will significatly increase their productivity.
+
+## Guide-level Explanation (aka Product Details)
+
+### New Metadata Definitions
+#### Semantic Metadata
+Metadata related to the feature itself.
+- Key: what makes a feature uniquely identifiable is the group, name and version. We don’t use db/source because the same feature can exists in different places (like hive and dynamo)
+    - ex: test_feature_group_name/test_feature_name/1.2.0
+- Availability/sources: list of places where feature is available
+- Name: the name of the feature
+- Version: the version of the feature
+- Data type
```

Comment:
could you add some examples of what data type for feature you have in mind?

Reply:
some examples for us are bigint, string, map but since it is represented as a string it can be anything